program/src/state/config.rs

```diff
@@ -0,0 +1,52 @@
+use bytemuck::{Pod, Zeroable};
+use shank::ShankAccount;
+use solana_program::pubkey::Pubkey;
+use spl_discriminator::SplDiscriminate;
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+
+/// Configuration for a staking system.
+#[repr(C)]
+#[derive(Clone, Copy, Pod, ShankAccount, SplDiscriminate, Zeroable)]
+#[discriminator_hash_input("config::account")]
+pub struct Config {
+    /// Account disciminator.
+    ///
+    /// The discriminator is equal to `ArrayDiscriminator:: UNINITIALIZED` when
+    /// the account is empty, and equal to `Config::DISCRIMINATOR` when the account
+    /// is initialized.
+    pub discriminator: [u8; 8],
+
+    /// Authority that can modify any elements in the config.
+    // TODO: does this need to be optional?
+    pub authority: OptionalNonZeroPubkey,
+
+    /// Authority that can slash any stake.
+    // TODO: does this need to be optional?
+    pub slash_authority: OptionalNonZeroPubkey,
+
+    /// Token account storing all stake.
+    pub vault_token: Pubkey,
```

Comment:
I would maybe rename this to something else so consumers understand this is a vault _token account_ and not a mint.

Reply:
It was renamed to `vault` on the instructions, so I used the same name here.